compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1360,67 +1340,85 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
 
         let opaque_ty_def_id = self.local_def_id(opaque_ty_node_id);
 
-        let mut collected_lifetimes = FxHashMap::default();
+        let mut collected_lifetimes = Vec::new();
+        let mut new_remapping = FxHashMap::default();
```

Comment:
I think this hash-mapa can move into the call to `with_hir_id_owner`? it doesn't seem to be referenced in many places, maybe it can just be returned by `create_lifetime_defs` instead of being mutated?

Reply:
About this, I'm moving it next to `create_lifetime_defs`. It's better to mutate it because of how we use this function in the async case.